types/src/index.ts

```diff
@@ -349,34 +349,21 @@ interface BaseSource<
   run: (this: PropThis<SourcePropDefinitions> & Methods & EmitFunction, options?: SourceRunOptions) => void | Promise<void>;
 }
 
-export interface LastSource<
-  Methods,
-  SourcePropDefinitions
-> extends BaseSource<
-  Methods,
-  SourcePropDefinitions
-> {
-  dedupe?: "last";
-}
-
 export type DedupedSource<
   Methods,
   SourcePropDefinitions
 > = Modify<BaseSource<
   Methods,
   SourcePropDefinitions
 >, {
-  dedupe: "greatest" | "unique";
+  dedupe: "last" | "greatest" | "unique";
```

Comment:
@michelle0927 aren't the semantics different here, assuming that a deduping strategy is always given? The default is `last`, which is the same as not specifying any, which is a valid source definition. The changes here means that with these types there is no default anymore and a strategy has to be always defined. That's not how the actual code works though, I believe, as it would be a backwards-incompatible change to the runtime.